lib/proxy_api/resource.rb

```diff
@@ -84,5 +84,11 @@ def put(payload, path = "")
     def delete(path)
       resource[path].delete
     end
+
+    def cached_get(path = nil, payload = {})
+      Rails.cache.fetch("proxies/#{url}", :expires_in => 3.minutes) do
+        get(path, payload)
```

Comment:
I still think there are benefits in having this here. It's a separate method name so this is safe from misusage.

The current implementation will need to have separate caching code per particular function.